core/src/identity.rs

```diff
@@ -174,42 +173,37 @@ impl Keypair {
             }
         };
 
-        Ok(pk.encode_to_vec())
+        Ok(quick_protobuf::serialize_into_vec(&pk).expect("Encoding to succeed."))
```

Comment:
I believe this is infallible by looking at the [code](https://docs.rs/quick-protobuf/latest/src/quick_protobuf/writer.rs.html#311-319). Unfortunately, I have not found a concrete answer in the docs or issues.

Reply:
I believe so too. One error I could find is https://docs.rs/quick-protobuf/latest/src/quick_protobuf/writer.rs.html#379 which can't happen if the `Vec` is allocated correctly.

The other error path comes from lines like https://github.com/tafia/quick-protobuf/blob/d977371e05170a016f03a80512b2a925468e3a1a/quick-protobuf/examples/pb_rs/data_types.rs#L199 which end up being the same error case as above so I think we are good.

I believe this function could actually be made infallible in `quick-protobuf` but I am not sure.